### DISCUSSION TITLE
Tpot ssh on specific port

### DISCUSSION THREAD

**Discussion by ARTER616** [telekom-security/tpotce#1366] [Q&A]
I have a vps where very few ports are allowed, so I cannot connect via SSH on port 64295. How can I change configuration of tpot to be able to connect on another port (not 64295)?

**Comment by Ganesh-403** [comment]
@ARTER616 
By default, **T-Pot intentionally moves the SSH daemon from port 22 to port 64295** during installation. This is part of the installer's default configuration and is documented in the project README. :contentReference[oaicite:0]{index=0}

If your VPS provider only allows a small set of inbound ports, you have two practical options:

### Option 1 (Recommended): Change the host's SSH port

Since T-Pot only changes the **host's SSH daemon configuration**, you can change it back (or to any allowed port) by editing your SSH server configuration.

For example:

```bash
sudo nano /etc/ssh/sshd_config
```

Change:

```text
Port 64295
```

to an allowed port, for example:

```text
Port 2222
```

Then restart SSH:

```bash
sudo systemctl restart ssh
```

or

```bash
sudo systemctl restart sshd
```

(depending on your distribution).

Don't forget to update your firewall/security group accordingly.

### Option 2: Port forwarding

If your VPS provider supports it, you can also forward an externally allowed port to `64295`. This avoids modifying the SSH daemon itself.

For example:

```text
External: 2222  --->  Internal: 64295
```

This is often the simplest solution if your provider offers NAT or firewall-based port forwarding.

### Things to keep in mind

- Changing the SSH port **does not affect the honeypots**—it only changes how you administer the host.
- If you update T-Pot in the future, review the release notes, as installer or update scripts may restore some default configuration depending on the upgrade path. From what I can tell, there isn't a documented T-Pot configuration variable for changing the SSH management port; it's simply the host's SSH service that is reconfigured during installation. :contentReference[oaicite:1]{index=1}

One question:

- Which ports **are** allowed by your VPS provider (e.g. `22`, `2222`, `443`)? That will help determine whether changing `sshd_config` or using port forwarding is the better approach.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
